### PR TITLE
Use URITemplate syntax to declare query parameters, fix #97

### LINKED ISSUE #97: Use URI template syntax to declare query parameters
URI template provides a concise and powerful way of describing query parameters. We can omit query_parameter annotation in most of the cases to reduce boilerplate.

**Suggested solution**
Use query parameters described in URI template:
Instead of:
```
@winter.route_get('/my-url/')
@winter.query_parameter('q')
```

Write:
```
@winter.route_get('/my-url/{?q}')
```

The new method should be preferred. The old method should work as before to keep backward compatibility and for `map_to` option.
Query parameters arrays should be supported too.
Swagger generator should be updated to support the new feature.

**Additional context**
URI Template RFC: https://tools.ietf.org/html/rfc6570

### PR DESCRIPTION
**Pull request by pristupa**

### CODE REVIEW

File: winter/routing/routing.py [hunk 1]
```diff
@@ -14,8 +19,20 @@ def route(
...
-    return annotate(annotation)
+    route_annotation = RouteAnnotation(url_path, http_method, produces, consumes)
+    query_annotations = get_query_param_annotations(url_path)
+
+    def wrapper(func_or_method):
+        annotate_ = annotate(route_annotation, single=True)
+        func_or_method = annotate_(func_or_method)
+
+        for query_annotation in query_annotations:
+            annotate_ = annotate(query_annotation, unique=True)
```
Comment (mofr) [inline winter/routing/routing.py]: I think that it's not very obvious behavior to artificially add annotations to methods. What do you think? Can we use it in arguments resolvers and inspectors only?

File: winter/core/annotation_decorator.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 class annotate_base(abc.ABC):
     _supported_classes = ()
 
-    def __init__(self, annotation: typing.Any, *, unique=False, single=False):
+    def __init__(self, annotation: typing.Any, unique=False, single=False):
```
Comment (mofr) [inline winter/core/annotation_decorator.py]: What is the reason for the change?

Comment (pristupa) [reply winter/core/annotation_decorator.py]: accidentally, will return back
